lib/itk/queue/consumer.ex

```diff
@@ -18,7 +28,9 @@ defmodule ITKQueue.Consumer do
 
   @doc false
   def init(subscription = %Subscription{}) do
-    subscribe(subscription)
+    Process.flag(:trap_exit, true)
```

Comment:
After adding `:trap_exit` here, shouldn't we listen for EXIT message as well in another `handle_info`?

Reply:
I'm not sure what you mean by *other* `handle_info`? We are reconnecting in the handler for `:DOWN` message.

Reply:
I meant that `:trap_exit` would be sending an `:exit` signal, we should have another function that expects `:exit` other than `:down` that reconnects as well.

Reply:
There's no need to handle exit messages, we cleanup in `terminate`.